core/test/base/range_accessors.cpp

```diff
@@ -121,4 +142,896 @@ TEST_F(RowMajorAccessor, CanAssignSubranges)
+    const gko::dim<3> size{4u, 3u, 2u};
+    static constexpr gko::size_type data_elements{4 * 3 * 2};
+    // clang-format off
+    st_type data[data_elements] {
+        // 0, y, z
+        1.0, 2.01,
+        -1.02, 3.03,
+        4.04, -2.05,
+        // 1, y, z
+        5.06, 6.07,
+        2.08, 3.09,
+        -1.1, -9.11,
+        // 2, y, z
+        -2.12, 2.13,
+        0.14, 15.15,
+        -9.16, 8.17,
+        // 3, y, z
+        7.18, -6.19,
+        5.2, -4.21,
+        3.22, -2.23
+    };
+    // clang-format on
+    reduced_storage r{size, data};
+    const_reduced_storage cr{size, data};
+
+    template <typename Accessor>
+    static void check_accessor_correctness(
+        const Accessor &a, std::tuple<int, int, int> ignore = t(99, 99, 99))
+    {
+        // Test for equality is fine here since they should not be modified
+        // clang-format off
+        if (ignore != t(0, 0, 0)) { EXPECT_EQ(a(0, 0, 0), st_type{1.0});     }
+        if (ignore != t(0, 0, 1)) { EXPECT_EQ(a(0, 0, 1), st_type{2.01});    }
+        if (ignore != t(0, 1, 0)) { EXPECT_EQ(a(0, 1, 0), st_type{-1.02});   }
+        if (ignore != t(0, 1, 1)) { EXPECT_EQ(a(0, 1, 1), st_type{3.03});    }
+        if (ignore != t(0, 2, 0)) { EXPECT_EQ(a(0, 2, 0), st_type{4.04});    }
+        if (ignore != t(0, 2, 1)) { EXPECT_EQ(a(0, 2, 1), st_type{-2.05});   }
+        if (ignore != t(1, 0, 0)) { EXPECT_EQ(a(1, 0, 0), st_type{5.06});    }
+        if (ignore != t(1, 0, 1)) { EXPECT_EQ(a(1, 0, 1), st_type{6.07});    }
+        if (ignore != t(1, 1, 0)) { EXPECT_EQ(a(1, 1, 0), st_type{2.08});    }
+        if (ignore != t(1, 1, 1)) { EXPECT_EQ(a(1, 1, 1), st_type{3.09});    }
+        if (ignore != t(1, 2, 0)) { EXPECT_EQ(a(1, 2, 0), st_type{-1.1});    }
+        if (ignore != t(1, 2, 1)) { EXPECT_EQ(a(1, 2, 1), st_type{-9.11});   }
+        if (ignore != t(2, 0, 0)) { EXPECT_EQ(a(2, 0, 0), st_type{-2.12});   }
+        if (ignore != t(2, 0, 1)) { EXPECT_EQ(a(2, 0, 1), st_type{2.13});    }
+        if (ignore != t(2, 1, 0)) { EXPECT_EQ(a(2, 1, 0), st_type{0.14});    }
+        if (ignore != t(2, 1, 1)) { EXPECT_EQ(a(2, 1, 1), st_type{15.15});   }
+        if (ignore != t(2, 2, 0)) { EXPECT_EQ(a(2, 2, 0), st_type{-9.16});   }
+        if (ignore != t(2, 2, 1)) { EXPECT_EQ(a(2, 2, 1), st_type{8.17});    }
+        if (ignore != t(3, 0, 0)) { EXPECT_EQ(a(3, 0, 0), st_type{7.18});    }
+        if (ignore != t(3, 0, 1)) { EXPECT_EQ(a(3, 0, 1), st_type{-6.19});   }
+        if (ignore != t(3, 1, 0)) { EXPECT_EQ(a(3, 1, 0), st_type{5.2});     }
+        if (ignore != t(3, 1, 1)) { EXPECT_EQ(a(3, 1, 1), st_type{-4.21});   }
+        if (ignore != t(3, 2, 0)) { EXPECT_EQ(a(3, 2, 0), st_type{3.22});    }
+        if (ignore != t(3, 2, 1)) { EXPECT_EQ(a(3, 2, 1), st_type{-2.23});   }
+        // clang-format on
+    }
+};
+
+using ReducedStorage3dTypes =
+    ::testing::Types<std::tuple<double, double>, std::tuple<double, float>,
+                     std::tuple<float, float>
```

Comment:
Quick question here, our custom float16/half implementation doesn't work for this? I believe it was made to work in combination with float, and to be used purely as a reduced storage.

Reply:
It should work here. Only the complex versions currently do not work because the ASSERT_NEAR don't work with complex.
I made it to work, just need to dial in the delta value.